src/main/java/BasicMAPF/Solvers/CBS/CBS_Solver.java

```diff
@@ -178,7 +188,14 @@ private void initOpen(I_ConstraintSet initialConstraints) {
      * Creates a root node.
      */
     private CBS_Node generateRoot(I_ConstraintSet initialConstraints) {
-        Solution solution = new Solution(); // init an empty solution
+        Solution solution; // init an empty solution
+        if (transientMAPFBehaviour.isTransientMAPF()) {
+            solution = new TransientMAPFSolution();
+        }
+        else {
+            solution = new Solution();
+        }
+
```

Comment:
```suggestion
        // init an empty solution
        Solution solution = transientMAPFBehaviour.isTransientMAPF() ? new TransientMAPFSolution() : new Solution(); 
```
just a little cleaner